web/src/engine/predictive-text/worker-thread/src/main/correction/distance-modeler.ts

```diff
@@ -157,68 +258,232 @@ export class SearchNode {
   }
 
   /**
-   * Adds paths that delete the next Transform from the input to better match words from
-   * the lexicon.  This aims to model (and ignore) when the user accidentally double-taps
-   * an input key.
+   * This method is used while stepping through intermediate deletion 'edges'
+   * for transforms with multi-character insert strings.
+   * @returns
+   */
+  private processDeletionSubset(): SearchNode {
+    // We're on easy street:  all transforms are already essentially merged into
+    // a single mass here.
+    let calculation = this.calculation.addInputChar({ key: SENTINEL_CODE_UNIT });
+
+    // If on a 'delete' style path, we just build out the paths into a single
+    // merged path.
+    const node = new SearchNode(this);
+    node.calculation = calculation;
+    // no update to the match string or traversal to be found here...
+    node.partialEdge.subsetSubindex++;
+    return node;
+  }
+
+  /**
+   * Finalizes the results of search nodes that represent the last step for
+   * processing multi-character insert transforms.
+   * @returns
+   */
+  private tryFinalize() {
+    const subset = this.partialEdge?.transformSubset;
+    if(!subset || subset.key > this.partialEdge.subsetSubindex) {
+      // Not yet ready for finalization.  Just exit.
+      return this;
+    }
+
+    // Finalization time!  We can safely transition the result node out
+    // of 'subset' mode.
+    delete this.partialEdge;
+
+    // Whatever entries are in the subset, they actually resolve down to the
+    // same net edit, as specified here.  It's more efficient to build the
+    // transform insert string on the subset, rather than mass-editing a group
+    // at each step and then consolidating them at the end.
+    this.priorInput.push({
+      sample: { insert: subset.insert, deleteLeft: subset.entries[0]?.sample.deleteLeft ?? 0 },
```

Comment:
Noting this as the answer to https://github.com/keymanapp/keyman/pull/14581/files#r2301097783.